### ISSUE TITLE
Opensearch icon support is crippled

### ISSUE THREAD

**Issue by SFITCS** [splitbrain/dokuwiki#1694]
DokuWiki will only produce valid image data for the Opensearch xml file if the source favicon.ico is composed of a 16x16.png, a 32x32.png, and a 48x48.png, and if those images are 8-bit. Otherwise it creates new lines in the base64 data (at 76 characters) and leaves off the last character - which means the OpenSearch browser entry will not have an icon.

A favicon.ico with more than three images, of different sizes, or 24-bit (or mixture of 8 and 24-bit) will display fine as a both a favicon or an icon for OpenSearch.

If an OpenSearch xml file is manually created and added to the browser it will display fine (i.e. the problem is in DokuWiki).

As it stands, if you want to use an icon for Opensearch you can't use large favicons, or more than 8-bit.

Attached contains the default dokuwiki favicon.ico and the valid Opensearch xml it creates. A custom favicon.ico containing more than three image size, which are not all 8-bit, a valid manually created Opensearch xml file. And also an example of a broken Opensearch.xml generated by DokuWiki.
[Examples.tar.gz](https://github.com/splitbrain/dokuwiki/files/480151/Examples.tar.gz)

NOTE: I couldn't find any documentation on this limitation to DokuWiki and Opensearch, and it took a bit of (frustrating) experimentation to find out why custom favicons wouldn't display icons in Opensearch.

 I don't have write access to DokuWiki documentation, but I've written a guide in a [forum post](https://forum.dokuwiki.org/thread/14062) for anyone who is motivated and does.  Thanks.

**Comment by Klap-in** [comment]
Is this behaviour happening for all browsers?

A quite random google search hits at 10kB limit for remote linked icons. Your case is about embedded, so this limit should not apply.
https://developer.mozilla.org/en-US/Add-ons/Creating_OpenSearch_plugins_for_Firefox

The head page of the FAQ is protected to prevent that everybody address there every question and against people who spam this page. btw the pages in the `faq:` namespace are editable by everyone. Now questions that are really asked regularly are promoted to that page. Further this section is to address specific DokuWiki issues, your question is a more specific customizing question. A nice one for the `tips:` namespace.

**Comment by SFITCS** [comment]
@Klap-in 
Yes. Tested with desktop browsers: Firefox current and ESR, IE8, and current Chrome/Chromium.

Agreed about the 10KB limit only applying to URI:http and not base64 encoded

The [Best Practice guide](http://www.opensearch.org/Documentation/Developer_best_practices_guide) recommends that a [64x64 pixel image](http://www.opensearch.org/Specifications/OpenSearch/1.1#The_.22Image.22_element) be included in the favicon, which DokuWiki currently does not support. When it is included (see previously attached archive containing _valid_manually_created.xml_) Firefox will use the 64x64 pixel image for both the page tag and the Opensearch icon. I haven't tested larger images (yet).

As for FAQ: I'll add something to the tips namespace soon, though I note that [Google](https://www.google.com.au/search?q=favicon+site:forums.dokuwiki.org) reflects my own experience - that there has been a lot of posts seeking information about favicons in DokuWiki (though I'm not sure how many of those posters even glance at documentation before posting).
